esmvaltool/preprocessor/__init__.py

```diff
@@ -47,8 +53,14 @@
     'mask_fillvalues',
     # Region selection
     'extract_region',
+    'extract_volume',
+    'extract_trajectory',
+    'extract_transect',
     # Grid-point operations
+    'depth_integration',
     'average_region',
+    'average_volume',
+    'time_average',
```

Comment:
For the reasons outlined in the discussion, I think this should be moved after `multi_model_statistics`, to keep the processing chain as general as possible.

Reply:
the `_multimodel` module should work fine with a single time point but will otherwise fail if there is no time axis. I agree with @mattiarighi on this one for generality purposes; having said this, if the multimodel statistics is intended to be computed for a time-averaged model ensemble (whether it be from daily data to monthly averages, from monthly data to yearly averages etc) then it is probably desirable to apply the time operation first and keep the time axis with a certain number of points and then apply the multimodel computation, that would save a lot of resources

Reply:
I agree that resources could be an issue, but it's conceptually not correct to apply time/area statistics before the multi-model. This is because, in general, the multi-model of some stats is not equal to the stats of the multi-model. 

Since the multi-model-mean/median is treated like another dataset, it's important that the preprocessing is applied consistently to all of them.

That's why I'm insisting on keeping the order of operations as originally planned. I understand that some diagnostic developers (like @ledm in this PR) would like to customize this order, but as I said this is not yet support and until then we need to stay with the most general approach.